lib/nrf_modem_lib/trace_backends/rtt/rtt.c

```diff
@@ -14,13 +14,15 @@ LOG_MODULE_REGISTER(modem_trace_backend, CONFIG_MODEM_TRACE_BACKEND_LOG_LEVEL);
 static int trace_rtt_channel;
 static char rtt_buffer[CONFIG_NRF_MODEM_LIB_TRACE_BACKEND_RTT_BUF_SIZE];
 
-int trace_backend_init(void)
+static trace_backend_processed_cb trace_processed_callback;
+
+int trace_backend_init(trace_backend_processed_cb trace_processed_cb)
 {
-#if defined(CONFIG_NRF_MODEM_LIB_TRACE_THREAD_PROCESSING)
 	const int segger_rtt_mode = SEGGER_RTT_MODE_BLOCK_IF_FIFO_FULL;
-#else
-	const int segger_rtt_mode = SEGGER_RTT_MODE_NO_BLOCK_SKIP;
-#endif
+
+	if (trace_processed_cb == NULL) {
+		return -EFAULT;
+	}
 
```

Comment:
We should add a check here if the trace backend is already initialized and return 0.
Now we allocate a new buffer on backend re-init. However, we don't (and should not) free the old one. 
```suggestion
        if (trace_processed_callback) {
                trace_processed_callback = trace_processed_cb;
                return 0;
       }
```

Reply:
I don't really understand this point

Reply:
Discussed offline. We should not allocate the RTT buffer again. Suggested change might not be the best so @MirkoCovizzi will make another.